Shareableviz Docs for Azure and GCP (#1748)
kedro-org/kedro-viz: docs/source/share_kedro_viz.md

# Publish and share Kedro-Viz

The publish and share feature on Kedro-Viz enables seamless sharing of pipeline visualizations on any static website hosting platform. There are two methods:

> 1. **Platform-agnostic sharing with Kedro-Viz:** Use the `kedro viz build` command to publish Kedro-Viz to platforms like GitHub pages, sharing the generated URL. This is described in [Platform-agnostic sharing with Kedro-Viz](./platform_agnostic_sharing_with_kedro_viz)

Author: [Nit] Use the `kedro viz build` command to publish Kedro-Viz **on** platforms

> # Publish and Share Kedro-Viz

Reviewer: ```suggestion
# Publish and share Kedro-Viz
```

Kedro users want to share their pipeline visualisation with other non-technical  stakeholders on their team. 

> Kedro enables users to do this by publishing Kedro-Viz to a static website hosting platform to share it with others. This guide uses the spaceflights tutorial as an example.

Author: I feel the intro is a bit more like 2 statements (line 3,5) which are not combined properly here. I am not able to think of anything right now but I feel the intro can be improved. any ideas @rashidakanchwala

> There are two ways to publish and share Kedro-Viz:

Reviewer: I found it a bit confusing that this says there are two ways of sharing Viz and then below there's also the header "Publish and share via CLI", so is this a third way or is it part of one of the other two?

```bash
pip install -r src/requirements.txt
```

Kedro-Viz requires specific minimum versions of `fsspec[s3]`, and `kedro` to publish your project. Ensure you have these correct versions by updating the `requirements.txt` file of the Kedro project to add the following:

```text
fsspec[s3]>=2023.9.0
kedro>=0.18.2
```

There are two ways to publish and share Kedro-Viz:

1. **Platform-agnostic sharing with Kedro-Viz:** Using the kedro-viz build command, users can publish Kedro-Viz to any static website hosting platform such as GitHub pages, and share the URL generated. This is described in [Platform-agnostic sharing with Kedro-Viz](./platform_agnostic_sharing_with_kedro_viz)

2. **Publish and share Kedro-Viz automatically:** You can automate the publishing and sharing of your Kedro-Viz through three different cloud providers: AWS, Azure, and GCS. Detailed instructions for each provider are outlined in the sections below.

* [Publish and share Kedro-Viz on AWS](./publish_and_share_kedro_viz_on_aws)
* [Publish and share Kedro-Viz on Azure](./publish_and_share_kedro_viz_on_azure)
* [Publish and share Kedro-Viz on GCP](./publish_and_share_kedro_viz_on_gcp)

```{note}

> From Kedro-Viz version 8.0.0, in addition to AWS, you will be able to publish and share your Kedro-Viz on Azure and GCP.

Author: May be we can remove this note from here and keep it in cloud provider specific pages like for Azure - From Kedro-Viz version 8.0.0, you will be able to publish and share your Kedro-Viz on Azure.

Reviewer 2: Resolved.

> When prompted for a project name, you can enter anything, but we will assume `Spaceflights` throughout.

Reviewer 3: I think this whole section shouldn't be a part of the intro page anymore. But each tutorial page should start off with this section similar to how the 'Preview Datasets' docs look

This page introduces the feature and different ways to use it  - https://kedro--1757.org.readthedocs.build/projects/kedro-viz/en/1757/preview_datasets.html
This page contains the guide including how to set up spaceflights project - https://kedro--1757.org.readthedocs.build/projects/kedro-viz/en/1757/preview_matplotlib_datasets.html

Reviewer 2: @rashidakanchwala done!

When your project is ready, navigate to the root directory of the project. Install the dependencies from the project root directory by typing the following in your terminal:

```bash
pip install -r src/requirements.txt
```

Kedro-Viz requires specific minimum versions of `fsspec[s3]`, and `kedro` to publish your project. Ensure you have these correct versions by updating the `requirements.txt` file of the Kedro project to add the following:

```text
fsspec[s3]>=2023.9.0
kedro>=0.18.2
```

There are two ways to publish and share Kedro-Viz:

> 1. **Platform-agnostic sharing with Kedro-Viz:** Using the kedro-viz build command, users can publish Kedro-Viz to any static website hosting platform such as GitHub pages, and share the URL generated. This is described in [Platform-agnostic sharing with Kedro-Viz](./platform_agnostic_sharing_with_kedro_viz)

Reviewer 2: Thanks @merelcht for the review.

- Capitalisation was used for IAM user as that is the unique name used by AWS (https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users.html). We've retained it so as not to confuse users setting up their S3 bucket.

- The section on 'Billing' was added as it was a common user question and barrier  to adopting shareable Viz highlighted during user testing: https://github.com/kedro-org/kedro-viz/issues/1533

When prompted for a project name, you can enter anything, but we will assume `Spaceflights` throughout.

When your project is ready, navigate to the root directory of the project. Install the dependencies from the project root directory by typing the following in your terminal:

```bash
pip install -r src/requirements.txt
```

Kedro-Viz requires specific minimum versions of `fsspec[s3]`, and `kedro` to publish your project. Ensure you have these correct versions by updating the `requirements.txt` file of the Kedro project to add the following:

```text
fsspec[s3]>=2023.9.0
kedro>=0.18.2
```

There are two ways to publish and share Kedro-Viz:

1. **Platform-agnostic sharing with Kedro-Viz:** Using the kedro-viz build command, users can publish Kedro-Viz to any static website hosting platform such as GitHub pages, and share the URL generated. This is described in [Platform-agnostic sharing with Kedro-Viz](./platform_agnostic_sharing_with_kedro_viz)

> 2. **Publish and share Kedro-Viz automatically:** You can automate the process of publishing and sharing your Kedro-Viz through AWS. Once the storage account and credentials are configured, users can then generate a shareable link from the Kedro-Viz UI using the 'Publish and share' button.

Reviewer 3: ```suggestion
2. **Publish and share Kedro-Viz automatically:** You can automate the publishing and sharing of your Kedro-Viz through three different cloud providers: AWS, Azure, and GCS. Detailed instructions for each provider are outlined in the sections below.
```

> ```{note}

Reviewer 3: can this not be after the entire point 2.

From Kedro-Viz version 7.2.0, in addition to AWS, you will be able to publish and share your Kedro-Viz on Azure and GCP.
```

* [Publish and share Kedro-Viz on AWS](./publish_and_share_kedro_viz_on_aws)
* [Publish and share Kedro-Viz on Azure](./publish_and_share_kedro_viz_on_azure)
* [Publish and share Kedro-Viz on GCP](./publish_and_share_kedro_viz_on_gcp)

> ```{toctree}

Reviewer 3: this should be hidden.

> ```{toctree}

Reviewer 3: ```{toctree}
:maxdepth: 1
:hidden:
platform_agnostic_sharing_with_kedro_viz
publish_and_share_kedro_viz_on_aws
publish_and_share_kedro_viz_on_azure
publish_and_share_kedro_viz_on_gcp
```

> ```{toctree}

Reviewer 3: ```suggestion
```{toctree}
:maxdepth: 1
:hidden:
platform_agnostic_sharing_with_kedro_viz
publish_and_share_kedro_viz_on_aws
publish_and_share_kedro_viz_on_azure
publish_and_share_kedro_viz_on_gcp
```
```

> :caption: Learn about Publish and Share Kedro-Viz

Reviewer 3: all of this should be hidden since we mention it above.

:maxdepth: 2

platform_agnostic_sharing_with_kedro_viz
publish_and_share_kedro_viz_on_aws
publish_and_share_kedro_viz_on_azure
publish_and_share_kedro_viz_on_gcp
```

> ## Publish and share via CLI

Reviewer 3: Can this be also part of the cloud provider page.

Author: yes even I was in two minds about this. Since CLI command is common to all cloud providers, I mentioned it here. But even for running the CLI command, the setup needs to be done specific to cloud provider. May be we can move this to cloud provider specific page

When your project is ready, navigate to the root directory of the project.

## Publish and share Kedro-Viz automatically

There are two ways to publish and share your Kedro-Viz:

> 1. You can automate the process of publishing and sharing your Kedro-Viz. This section describes the steps for AWS, which is the only cloud provider supported for automation at present. Integration with other cloud providers, namely Azure and GCP, will be added soon.

Reviewer 2: Do we need to update this since we are adding Azure and GCP?

Author: I added a note for 7.2.0 separately as I was not sure how to describe for users using viz sharing before 7.2.0
